src/Chainweb/Miner/Coordinator.hs

```diff
@@ -121,18 +122,27 @@ newtype PrevTime = PrevTime BlockCreationTime
 
 data ChainChoice = Anything | TriedLast ChainId | Suggestion ChainId
 
+-- | A `Miner` defined in the nodes `miners` list. Should be subject to Primed
+-- Coordination.
+--
+newtype Whitelisted = Whitelisted Miner
```

Comment:
I think the code below in this PR would be more readable if this would be something like:

```Haskell
data MiningClient = Primed Miner | Public Miner

miningClient :: MiningClient -> Miner
miningClient (Primed m) = m
miningClient (Public m) = m
```

instead of passing around `Either Miner Whitelisted` (which is isomorphic to above type but less explicit and more complex).